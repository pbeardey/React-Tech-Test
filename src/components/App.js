// src/components/App.js

import React from "react";
import '../styles/app.css';
import Search from "./Search";

const App = () => {
  return (
    <div className="App">
        <img className="nasa-logo" alt="NASA logo" src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"/>
        <Search />
    </div>
  );
}

export default App;
