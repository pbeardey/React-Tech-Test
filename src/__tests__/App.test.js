// src/__tests__/App.test.js

import { render, screen } from '@testing-library/react';
import App from '../components/App';

describe("App", () => {
  
  it("renders App", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders logo", () => {
    render(<App />);
    const logoAltText = screen.getByAltText(/NASA logo/i);
    expect(logoAltText).toBeInTheDocument();
  })
});