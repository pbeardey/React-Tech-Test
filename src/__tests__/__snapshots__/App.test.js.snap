// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <img
      alt="NASA logo"
      class="nasa-logo"
      src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
    />
    <input
      class="search-input"
      type="text"
    />
  </div>
</DocumentFragment>
`;
